mdpow/analysis/workflows/base.py

```diff
@@ -0,0 +1,156 @@
+#import will need to change based on new naming convention for ada
+#and once the function changes for more general use
+import mdpow.analysis.workflows.dihedrals as ada
+
+import logging
+
+logger = logging.getLogger('mdpow.analysis.workflows.base')
+
+def directory_paths(parent_directory=None, csv=None):
+    '''
+       .. function:: base.directory_paths
+       Takes a parent directory containing MDPOW simulation project subdirectories,
+       or .csv file containing :code:`molname`, :code:`resname`, and
+       simulation data directory paths as argument and returns a
+       :class:`pandas.DataFrame` for use with
+       :func:`~mdpow.analysis.workflows.base.directory_iteration`, which iterates
+       :func:`~mdpow.analysis.workflows.dihedrals.automated_dihedral_analysis`
+       over the project directories included in the
+       :func:`~mdpow.analysis.workflows.base.directory_paths` :class:`pandas.DataFrame`.
+       
+       :keywords:
+       
+       *parent_directory*
+           the path for the location of the top directory 
+           under which the subdirectories of MDPOW simulation
+           data exist, additionally creates a 'dir.csv' file
+           for user manipulation of metadata and future reference
+           
+       *csv*
+           .csv file containing the molecule names, resnames,
+           and paths, in that order, for the MDPOW simulation
+           data to be iterated over
+           must contain header of the form:
+           format: molecule,resname,path
+           
+       .. rubric:: Examples
+       
+           directory_paths = directory_paths(parent_directory='/foo/bar/MDPOW_projects')
+           directory_iteration(directory_paths)
+           
+           or
+           
+           directory_paths = directory_paths(csv='/foo/bar/MDPOW.csv')
+           directory_iteration(directory_paths)
+           
+    '''
+
+    if parent_directory is not None:
+
+        locations = []
+
+        reg_compile = re.compile('FEP')              
+        for dirpath, dirnames, filenames in os.walk(parent_directory):
+            result = [dirpath.strip() for dirname in dirnames if  reg_compile.match(dirname)]
+            if result:
+                locations.append(result[0])
+
+        resnames = []
+
+        for loc in locations:
+            res_temp = loc.strip().split('/')
+            resnames.append(res_temp[-1])
+
+        directory_paths = pd.DataFrame(
+        {
+            "molecule": resnames,
+            "resname": resnames,
+            "path": locations
+        }
+    )
+        directory_paths.to_csv('dir.csv', index=False)
+
+    elif csv is not None:
+        locations = pd.read_csv(csv)
+        directory_paths = locations.sort_values(by=['molecule', 'resname', 'path']).reset_index(drop=True)
+
+    return directory_paths
+
+#needs to be changed for use with other automated workflows
+def directory_iteration(directory_paths, df_save_dir=None, figdir=None, padding=45, width=0.9,
```

Comment:
This should be made more general so that you can specify which analysis to take. Reduce the code to the generic bits.

Reply:
This will require adaptation of each analysis for use with iteration and desired outcomes (plotting and storing data), which might require a separate workflows module for each (the automation class I made does not make sense because the analyses are not similar enough for significant overlap, it would only confuse those who want to develop their own analyses from the base class). I will start new pull requests to make these changes for each once this pull request is merged and I have more of an idea of how to generalize this function for use with all of the automated physical properties analyses, including those that users might develop on their own.